cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -267,8 +301,12 @@ func logServiceAddedAccordingToMavenDependencyAndExtraCondition(
 func detectSpringBootVersion(currentRoot *mavenProject, mavenProject *mavenProject) string {
 	// mavenProject prioritize than rootProject
 	if mavenProject != nil {
-		return detectSpringBootVersionFromProject(mavenProject)
-	} else if currentRoot != nil {
+		if version := detectSpringBootVersionFromProject(mavenProject); version != UnknownSpringBootVersion {
```

Comment:
bugfix: when spring boot version not detected in the current project, should continue to detect in the root project, instead of return.

Reply:
How about updating the function of `detectSpringBootVersionFromProject`, and set both `mavenProject` and `currentRoot` as parameter?

Reply:
Could you please add unit test for this case?

Reply:
Updated in another PR.